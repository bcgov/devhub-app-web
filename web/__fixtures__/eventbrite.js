/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

// event brite sample data

const EVENT_BRITE_EVENT = {
  internal: {
    content:
      '{"name":{"text":"Lunch & Lab - How to Safely Navigate Open Source Licensing in your Projects","html":"Lunch &amp; Lab - How to Safely Navigate Open Source Licensing in your Projects"},"description":{"text":"WHEN? Friday, April 6, 12-1 pm. CSI Lab Commons, 1012 Douglas. 3rd floor. Come hungry. There will be snacks. Everyone welcome.\\nYou have an existing software application and you want to “crowd source” a fix/update. Maybe you have a new application build that will benefit from leveraging existing open source software, or maybe all you have is an idea in need of a solution. Whether the goal is to make existing government-owned software available under an open source license or bring open source code in, this talk will help you understand what you need to know to comply with government policies for success with your project. We will also be sharing some “lessons learned” and field any questions you may have.\\nBio: Lori Kernaghan is an Intellectual Property Manager with the Intellectual Property Program, Ministry of Citizens’ Services. She has 15 years’ experience as an intellectual property professional. Lori joined the BC Government in 2008 and for the past 10 years has been providing ministry and core agency clients across government with a variety of intellectual property management services, including licensing of intellectual property. Lori has worked on a number of key initiatives in the BC Government, including the Open Information and Open Data Policy, the BC Open Source Development Employee Guide (GitHub) and the development of social media guidelines. \\nRecommended (but optional) Pre-Reading: BC Policy Framework For GitHub\\nWhat is the CSI Lab?\\nThe CSI Lab is a creative space where multi-disciplinary teams of public servants can learn new ways of working using technology as an enabler to build a modern public service.","html":"<P><STRONG>WHEN? Friday, April 6, 12-1 pm. CSI Lab Commons, 1012 Douglas. 3rd floor. Come hungry. There will be snacks. Everyone welcome.</STRONG></P>\\n<P>You have an existing software application and you want to “crowd source” a fix/update. Maybe you have a new application build that will benefit from leveraging existing open source software, or maybe all you have is an idea in need of a solution. Whether the goal is to make existing government-owned software available under an open source license or bring open source code in, this talk will help you understand what you need to know to comply with government policies for success with your project. We will also be sharing some “lessons learned” and field any questions you may have.</P>\\n<P><SPAN><STRONG>Bio</STRONG>:</SPAN> Lori Kernaghan is an Intellectual Property Manager with the Intellectual Property Program, Ministry of Citizens’ Services. She has 15 years’ experience as an intellectual property professional. Lori joined the BC Government in 2008 and for the past 10 years has been providing ministry and core agency clients across government with a variety of intellectual property management services, including licensing of intellectual property. Lori has worked on a number of key initiatives in the BC Government, including the Open Information and Open Data Policy, the BC Open Source Development Employee Guide (GitHub) and the development of social media guidelines. </P>\\n<P><SPAN><STRONG>Recommended (but optional) Pre-Reading:</STRONG> </SPAN><A HREF=\\"https://github.com/bcgov/BC-Policy-Framework-For-GitHub\\" TARGET=\\"_blank\\" REL=\\"nofollow noopener noreferrer\\">BC Policy Framework For GitHub</A></P>\\n<P><STRONG>What is the CSI Lab?</STRONG></P>\\n<P>The CSI Lab is a creative space where multi-disciplinary teams of public servants can learn new ways of working using technology as an enabler to build a modern public service.</P>"},"id":"44648171884","url":"https://www.eventbrite.ca/e/lunch-lab-how-to-safely-navigate-open-source-licensing-in-your-projects-tickets-44648171884","start":{"timezone":"America/Vancouver","local":"2018-04-06T12:00:00","utc":"2018-04-06T19:00:00Z"},"end":{"timezone":"America/Vancouver","local":"2018-04-06T13:00:00","utc":"2018-04-06T20:00:00Z"},"organization_id":"228490647317","created":"2018-03-28T18:27:40Z","changed":"2018-04-07T03:48:10Z","published":"2018-03-28T18:29:56Z","capacity":50,"capacity_is_custom":false,"status":"completed","currency":"CAD","listed":false,"shareable":false,"invite_only":false,"online_event":false,"show_remaining":false,"tx_time_limit":480,"hide_start_date":false,"hide_end_date":false,"locale":"en_CA","is_locked":false,"privacy_setting":"unlocked","is_series":false,"is_series_parent":false,"inventory_type":"limited","is_reserved_seating":false,"show_pick_a_seat":false,"show_seatmap_thumbnail":false,"show_colors_in_seatmap_thumbnail":false,"source":"create_2.0","is_free":true,"version":"3.0.0","summary":"WHEN? Friday, April 6, 12-1 pm. CSI Lab Commons, 1012 Douglas. 3rd floor. Come hungry. There will be snacks. Everyone welcome.\\nYou have an e","logo_id":"42845796","organizer_id":"17134258064","venue_id":"23970647","category_id":null,"subcategory_id":null,"format_id":null,"resource_uri":"https://www.eventbriteapi.com/v3/events/44648171884/","is_externally_ticketed":false,"logo":{"crop_mask":{"top_left":{"x":0,"y":0},"width":794,"height":397},"original":{"url":"https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?auto=compress&s=a2829f497244deb7ebb5fe91f5c9400d","width":794,"height":1123},"id":"42845796","url":"https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?h=200&w=450&auto=compress&rect=0%2C0%2C794%2C397&s=ae5a86a02c4ef24f4be03bdda60d27a1","aspect_ratio":"2","edge_color":null,"edge_color_set":true}}',
    description: 'DEMO EVENT BRITE',
    mediaType: 'text/html',
    type: 'EventbriteEvents',
  },
  name: {
    text: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
    html: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
  },
  description: {
    text: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
    html: 'DEMO EVENT BRITE DATA - DEVELOPMENT ONLY',
  },
  url:
    'https://www.eventbrite.ca/e/lunch-lab-how-to-safely-navigate-open-source-licensing-in-your-projects-tickets-44648171884',
  start: {
    timezone: 'America/Vancouver',
    local: '2018-04-06T12:00:00',
    utc: '2018-04-06T19:00:00Z',
  },
  end: {
    timezone: 'America/Vancouver',
    local: '2018-04-06T13:00:00',
    utc: '2018-04-06T20:00:00Z',
  },
  organizer_id: '17134258064',
  organization_id: '228490647317',
  created: '2018-03-28T18:27:40Z',
  changed: '2018-04-07T03:48:10Z',
  published: '2018-03-28T18:29:56Z',
  privacy_setting: 'unlocked',
  capacity: 50,
  category_id: null,
  currency: 'CAD',
  capacity_is_custom: false,
  locale: 'en_CA',
  shareable: true,
  logo: {
    id: 'DEMO',
    url:
      'https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?h=200&w=450&auto=compress&rect=0%2C0%2C794%2C397&s=ae5a86a02c4ef24f4be03bdda60d27a1',
    crop_mask: {
      width: 794,
      height: 397,
      top_left: {
        x: 0,
        y: 0,
      },
    },
    original: {
      url:
        'https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F42845796%2F228490647317%2F1%2Foriginal.jpg?auto=compress&s=a2829f497244deb7ebb5fe91f5c9400d',
      width: 794,
      height: 1123,
      localFile: {
        id: 'e87c2530-f76c-5e28-bf50-f4e483371cd0',
      },
    },
    aspect_ratio: '2',
    edge_color: null,
    edge_color_set: true,
  },
  logo_id: 'DEMO@',
  organization: 'CSI Lab',
  venue_id: 'DEMO_VENUE',
  format_id: null,
  resource_uri: 'https://www.eventbriteapi.com/v3/events/DEMO/',
  is_externally_ticketed: false,
  series_id: null,
  venue: {
    id: '23970647',
    address: {
      address_1: '3rd Floor, 1012 Douglas Street',
      address_2: null,
      city: 'Victoria',
      region: 'BC',
      postal_code: 'V8W 2C3',
      country: 'CA',
      latitude: '48.4242409',
      longitude: '-123.36581739999997',
      localized_address_display: '3rd Floor, 1012 Douglas Street, Victoria, BC V8W 2C3',
      localized_area_display: 'Victoria, BC',
    },
    resource_uri: 'https://www.eventbriteapi.com/v3/venues/23970647/',
    name: '1012 Douglas St',
    latitude: '48.4242409',
    longitude: '-123.36581739999997',
    internal: {
      type: 'EventbriteVenues',
      content:
        '{"address":{"address_1":"3rd Floor, 1012 Douglas Street","address_2":null,"city":"Victoria","region":"BC","postal_code":"V8W 2C3","country":"CA","latitude":"48.4242409","longitude":"-123.36581739999997","localized_address_display":"3rd Floor, 1012 Douglas Street, Victoria, BC V8W 2C3","localized_area_display":"Victoria, BC","localized_multi_line_address_display":["3rd Floor, 1012 Douglas Street","Victoria, BC V8W 2C3"]},"resource_uri":"https://www.eventbriteapi.com/v3/venues/23970647/","id":"23970647","age_restriction":null,"capacity":null,"name":"1012 Douglas St","latitude":"48.4242409","longitude":"-123.36581739999997"}',
      contentDigest: 'b72d19dba8d990adf5455d65ff6adae8',
      owner: 'gatsby-source-eventbrite',
    },
  },
};

module.exports = {
  EVENT_BRITE_EVENT,
};
