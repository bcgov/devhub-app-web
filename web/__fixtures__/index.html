<!-- DO NOT DELETE -->
<!-- This file is used for any future integration tests between the devhub and the react-rehype/component preview component -->
<!DOCTYPE html>
<html lang="en">
<head>
    <link href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.1.3/css/bootstrap-reboot.min.css" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Noto+Sans" rel="stylesheet">
    <style>
      header {
        background-color: blue;
        color: white;
        padding: 35px;
        border-bottom: 1px solid yellow;
      }
    </style>
    <title>Sample Header</title>
</head>

<!--
  All in-line CSS is specific to this sample; it can and should be ignored.
 -->

<body>
  <header>
    <div class="banner">
        <a href="https://gov.bc.ca" alt="British Columbia">
          <img src="../assets/images/logo-banner.svg" alt="Go to the Government of British Columbia website" />
        </a>
        <h1>Hello British Columbia</h1>
    </div>
    <div class="other">
    <!-- 
      This place is for anything that needs to be right aligned
      beside the logo.  
    -->
      &nbsp;
    </div>
    </div>
  </header>
</body>
</html>
