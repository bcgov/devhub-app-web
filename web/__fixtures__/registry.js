export const JOURNEY_REGISTRY = [
  {
    name: 'New Developer',
    sourceProperties: {
      stops: [
        {
          stops: [{ id: 10 }],
          id: 11,
        },
        {
          stops: [{ id: 13 }, { id: 14 }],
          id: 12,
        },
      ],
    },
  },
  {
    name: 'New Developer',
    sourceProperties: {
      stops: [
        {
          stops: [{ id: 2 }],
          id: 1,
        },
        {
          stops: [{ id: 4 }, { id: 5 }],
          id: 3,
        },
      ],
    },
  },
];

export const JOURNEY_REGISTRY_MAPPED_AS_TOPIC = [
  {
    name: 'New Developer',
    sourceProperties: {
      sources: [
        { id: 11 },
        {
          id: 10,
          connectsWith: {
            id: 11,
          },
        },
        {
          id: 12,
        },
        {
          id: 13,
          connectsWith: {
            id: 12,
          },
        },
        {
          id: 14,
          connectsWith: {
            id: 12,
          },
        },
      ],
    },
  },
  {
    name: 'New Developer',
    sourceProperties: {
      sources: [
        {
          id: 1,
        },
        {
          id: 2,
          connectsWith: {
            id: 1,
          },
        },
        {
          id: 3,
        },
        {
          id: 4,
          connectsWith: {
            id: 3,
          },
        },
        {
          id: 5,
          connectsWith: {
            id: 3,
          },
        },
      ],
    },
  },
];
