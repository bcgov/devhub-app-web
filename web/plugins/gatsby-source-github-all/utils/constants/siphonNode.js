/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

const { SOURCE_TYPES } = require('./sourceTypes');

const UNFURL_TYPES = {
  MARKDOWN: 'markdown',
  EXTERNAL: 'external',
};
// source types that map to fetch routines
const COLLECTION_TYPES = {
  [SOURCE_TYPES.GITHUB]: 'source',
  [SOURCE_TYPES.WEB]: 'source',
  CURATED: 'curated',
};

const RESOURCE_TYPES = {
  COMPONENTS: 'Components',
  DOCUMENTATION: 'Documentation',
  SELF_SERVICE_TOOLS: 'Self-Service Tools',
  REPOSITORIES: 'Repositories',
  PEOPLE: 'People',
  PROJECTS: 'Projects',
  EVENTS: 'Events',
};
// Resource Types for Devhub
const RESOURCE_TYPES_LIST = Object.values(RESOURCE_TYPES);

const PERSONAS = {
  DESIGNER: 'Designer',
  DEVELOPER: 'Developer',
  PRODUCT_OWNER: 'Product Owner',
};

const COLLECTION_TEMPLATES = {
  DEFAULT: 'default',
  OVERVIEW: 'overview',
};

const COLLECTION_TEMPLATES_LIST = Object.values(COLLECTION_TEMPLATES);

const PERSONAS_LIST = Object.values(PERSONAS);

const GRAPHQL_NODE_TYPE = {
  SIPHON: 'DevhubSiphon',
  COLLECTION: 'DevhubCollection',
};

module.exports = {
  UNFURL_TYPES,
  COLLECTION_TYPES,
  RESOURCE_TYPES,
  RESOURCE_TYPES_LIST,
  PERSONAS,
  PERSONAS_LIST,
  GRAPHQL_NODE_TYPE,
  COLLECTION_TEMPLATES,
  COLLECTION_TEMPLATES_LIST,
};
