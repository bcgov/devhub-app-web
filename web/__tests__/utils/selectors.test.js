import groupBy from 'lodash/groupBy';
import {
  selectTopicsWithResourcesGroupedByType,
  selectResourcesGroupedByType,
} from '../../src/utils/selectors';
import { RESOURCE_TYPES } from '../../src/constants/ui';
import { SIPHON_NODES, TOPICS } from '../../__fixtures__/nodes';

const defaultGroupings = Object.keys(RESOURCE_TYPES).reduce((obj, type) => {
  obj[RESOURCE_TYPES[type]] = [];
  return obj;
}, {});

describe('General Purpose Selectors', () => {
  it('selects resources grouped by type', () => {
    const selector = selectResourcesGroupedByType();
    const groupedResources = groupBy(SIPHON_NODES, 'resource.type');

    expect(selector(SIPHON_NODES)).toEqual({ ...defaultGroupings, ...groupedResources });
  });

  it('selects topics with resources (grouped by type) appeneded as a property', () => {
    const selector = selectTopicsWithResourcesGroupedByType();
    const topicWithGroupedResources = selector(TOPICS);

    expect(topicWithGroupedResources).toMatchSnapshot();
  });
});
