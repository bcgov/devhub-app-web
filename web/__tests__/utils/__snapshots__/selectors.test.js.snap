// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General Purpose Selectors selects topics with resources (grouped by type) appeneded as a property 1`] = `
Array [
  Object {
    "childrenDevhubSiphon": Array [
      Object {
        "_metadata": Object {
          "position": "000.000.002",
        },
        "id": "3",
        "resource": Object {
          "path": "/design-system/Z1bGIqu3",
          "type": "Components",
        },
        "unfurl": Object {
          "author": "ksingbeil",
          "data1": "",
          "data2": "",
          "description": "What the Design System is and how it works.",
          "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
          "label1": "",
          "label2": "",
          "title": "About",
          "type": "markdown",
        },
      },
      Object {
        "_metadata": Object {
          "position": "000.000.001",
        },
        "id": "2",
        "resource": Object {
          "path": "/design-system/Z1bGIqu2",
          "type": "Components",
        },
        "unfurl": Object {
          "author": "ksingbeil",
          "data1": "",
          "data2": "",
          "description": "What the Design System is and how it works.",
          "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
          "label1": "",
          "label2": "",
          "title": "About",
          "type": "markdown",
        },
      },
      Object {
        "_metadata": Object {
          "position": "000.000.000",
        },
        "id": "1",
        "resource": Object {
          "path": "/design-system/Z1bGIqu1",
          "type": "Components",
        },
        "unfurl": Object {
          "author": "ksingbeil",
          "data1": "",
          "data2": "",
          "description": "What the Design System is and how it works.",
          "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
          "label1": "",
          "label2": "",
          "title": "About",
          "type": "markdown",
        },
      },
      Object {
        "_metadata": Object {
          "position": "000.000.003",
        },
        "id": "4",
        "resource": Object {
          "path": "/design-system/Z1bGIqu4",
          "type": "Components",
        },
        "unfurl": Object {
          "author": "ksingbeil",
          "data1": "",
          "data2": "",
          "description": "What the Design System is and how it works.",
          "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
          "label1": "",
          "label2": "",
          "title": "About",
          "type": "markdown",
        },
      },
    ],
    "connectsWith": Array [
      Object {
        "_xxboundProperties": Object {
          "position": 1,
          "topicResourceType": "Documentation",
        },
        "childMarkdownRemark": Object {
          "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
        },
        "fields": Object {
          "author": "patricksimonian",
          "description": "Everything you need to know about working with Github in the BC Government.",
          "image": "../images/octokat.png",
          "pagePaths": Array [
            "/design-system/foo",
          ],
          "personas": Array [
            "Designer",
          ],
          "position": 1,
          "resourceType": "Documentation",
          "slug": "Introduction-To-Github-and-Gov",
          "standAlonePath": "/GitHub-Raw-1",
          "title": "Introduction To Github and Gov",
          "topics": Array [
            Object {
              "id": "topic-1",
            },
          ],
        },
        "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
        "name": "README.md",
        "path": "/Design-System/foo",
      },
      Object {
        "_xxboundProperties": Object {
          "position": 2,
          "topicResourceType": "Documentation",
        },
        "childMarkdownRemark": Object {
          "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
        },
        "fields": Object {
          "author": "patricksimonian",
          "description": "Everything you need to know about working with Github in the BC Government.",
          "image": "../images/octokat.png",
          "pagePaths": Array [
            "/design-system/bar",
          ],
          "personas": Array [
            "Designer",
          ],
          "position": 1,
          "resourceType": "Components",
          "slug": "Introduction-To-Github-and-Gov",
          "standAlonePath": "/GitHub-Raw-2",
          "title": "Introduction To Github and Gov",
          "topics": Array [
            Object {
              "id": "topic-2",
            },
          ],
        },
        "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
        "name": "README.md",
        "path": "/Design-System/bar",
      },
    ],
    "description": "baz",
    "fields": Object {
      "githubRaw": Array [
        Object {
          "_xxboundProperties": Object {
            "position": 1,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/foo",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Documentation",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-1",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-1",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
        },
        Object {
          "_xxboundProperties": Object {
            "position": 2,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/bar",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Components",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-2",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-2",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
        },
      ],
    },
    "hasResources": true,
    "id": "topic-1",
    "name": "Design System",
    "resources": Object {
      "Components": Array [
        Object {
          "_xxboundProperties": Object {
            "position": 2,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/bar",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Components",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-2",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-2",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
          "path": "/Design-System/bar",
        },
      ],
      "Documentation": Array [
        Object {
          "_xxboundProperties": Object {
            "position": 1,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/foo",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Documentation",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-1",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-1",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
          "path": "/Design-System/foo",
        },
      ],
      "Events": Array [],
      "Journey": Array [],
      "People": Array [],
      "Repositories": Array [],
      "Self-Service Tools": Array [],
      "Topics": Array [],
    },
    "type": "default",
  },
  Object {
    "childrenDevhubSiphon": Array [
      Object {
        "_metadata": Object {
          "position": "001.000.000",
        },
        "id": "5",
        "resource": Object {
          "path": "/design-system/Z1bGIqu5",
          "type": "Documentation",
        },
        "unfurl": Object {
          "author": "ksingbeil",
          "data1": "",
          "data2": "",
          "description": "What the Design System is and how it works.",
          "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
          "label1": "",
          "label2": "",
          "title": "About",
          "type": "markdown",
        },
      },
      Object {
        "_metadata": Object {
          "position": "001.000.001",
        },
        "id": "6",
        "resource": Object {
          "path": "/design-system/Z1bGIqu6",
          "type": "Documentation",
        },
        "unfurl": Object {
          "author": "ksingbeil",
          "data1": "",
          "data2": "",
          "description": "What the Design System is and how it works.",
          "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
          "label1": "",
          "label2": "",
          "title": "About",
          "type": "markdown",
        },
      },
      Object {
        "_metadata": Object {
          "position": "001.000.001",
        },
        "id": "7",
        "resource": Object {
          "path": "/design-system/Z1bGIqu7",
          "type": "Documentation",
        },
        "unfurl": Object {
          "author": "ksingbeil",
          "data1": "",
          "data2": "",
          "description": "What the Design System is and how it works.",
          "image": "https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true",
          "label1": "",
          "label2": "",
          "title": "About",
          "type": "markdown",
        },
      },
    ],
    "connectsWith": Array [
      Object {
        "_xxboundProperties": Object {
          "position": 1,
          "topicResourceType": "Documentation",
        },
        "childMarkdownRemark": Object {
          "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
        },
        "fields": Object {
          "author": "patricksimonian",
          "description": "Everything you need to know about working with Github in the BC Government.",
          "image": "../images/octokat.png",
          "pagePaths": Array [
            "/design-system/foo",
          ],
          "personas": Array [
            "Designer",
          ],
          "position": 1,
          "resourceType": "Documentation",
          "slug": "Introduction-To-Github-and-Gov",
          "standAlonePath": "/GitHub-Raw-1",
          "title": "Introduction To Github and Gov",
          "topics": Array [
            Object {
              "id": "topic-1",
            },
          ],
        },
        "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
        "name": "README.md",
        "path": "/Devhub/foo",
      },
      Object {
        "_xxboundProperties": Object {
          "position": 2,
          "topicResourceType": "Documentation",
        },
        "childMarkdownRemark": Object {
          "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
        },
        "fields": Object {
          "author": "patricksimonian",
          "description": "Everything you need to know about working with Github in the BC Government.",
          "image": "../images/octokat.png",
          "pagePaths": Array [
            "/design-system/bar",
          ],
          "personas": Array [
            "Designer",
          ],
          "position": 1,
          "resourceType": "Components",
          "slug": "Introduction-To-Github-and-Gov",
          "standAlonePath": "/GitHub-Raw-2",
          "title": "Introduction To Github and Gov",
          "topics": Array [
            Object {
              "id": "topic-2",
            },
          ],
        },
        "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
        "name": "README.md",
        "path": "/Devhub/bar",
      },
    ],
    "description": "baz",
    "fields": Object {
      "githubRaw": Array [
        Object {
          "_xxboundProperties": Object {
            "position": 1,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/foo",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Documentation",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-1",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-1",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
        },
        Object {
          "_xxboundProperties": Object {
            "position": 2,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/bar",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Components",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-2",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-2",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
        },
      ],
    },
    "hasResources": true,
    "id": "topic-2",
    "name": "Devhub",
    "resources": Object {
      "Components": Array [
        Object {
          "_xxboundProperties": Object {
            "position": 2,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/bar",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Components",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-2",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-2",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
          "path": "/Devhub/bar",
        },
      ],
      "Documentation": Array [
        Object {
          "_xxboundProperties": Object {
            "position": 1,
            "topicResourceType": "Documentation",
          },
          "childMarkdownRemark": Object {
            "id": "d72fa440-51f6-52c7-ab03-140907fd46db",
          },
          "fields": Object {
            "author": "patricksimonian",
            "description": "Everything you need to know about working with Github in the BC Government.",
            "image": "../images/octokat.png",
            "pagePaths": Array [
              "/design-system/foo",
            ],
            "personas": Array [
              "Designer",
            ],
            "position": 1,
            "resourceType": "Documentation",
            "slug": "Introduction-To-Github-and-Gov",
            "standAlonePath": "/GitHub-Raw-1",
            "title": "Introduction To Github and Gov",
            "topics": Array [
              Object {
                "id": "topic-1",
              },
            ],
          },
          "id": "8257de0d-811b-58dd-9fae-661f13d5a11b",
          "name": "README.md",
          "path": "/Devhub/foo",
        },
      ],
      "Events": Array [],
      "Journey": Array [],
      "People": Array [],
      "Repositories": Array [],
      "Self-Service Tools": Array [],
      "Topics": Array [],
    },
    "type": "default",
  },
]
`;
